File: resources/capability-bundle/.claude/skills/scenario-preparation/scripts/extract-material.mjs
import { readFile } from 'node:fs/promises';
import path from 'node:path';

const filePath = process.argv[2];
if (!filePath) throw new Error('Pass an authorized input file path.');

const extension = path.extname(filePath).toLowerCase();
const raw = await readFile(filePath, 'utf8');
let sections;

if (extension === '.json') {
  const parsed = JSON.parse(raw);
  sections = Object.entries(parsed).map(([heading, value]) => ({ heading, text: String(value) }));
} else {
  sections = raw
    .split(/^#{1,6}\s+/mu)
    .map((text, index) => ({ heading: index === 0 ? 'document' : `section-${index}`, text: text.trim() }))
    .filter((section) => section.text);
}

process.stdout.write(JSON.stringify({ source_name: path.basename(filePath), sections }));

File: resources/capability-bundle/.claude/skills/scenario-preparation/scripts/extract-material.py
import json
from pathlib import Path
import re
import sys


if len(sys.argv) < 2:
    raise ValueError("Pass an authorized input file path.")

file_path = Path(sys.argv[1])
raw = file_path.read_text(encoding="utf-8")
if file_path.suffix.lower() == ".json":
    parsed = json.loads(raw)
    sections = [
        {"heading": str(heading), "text": str(value)}
        for heading, value in parsed.items()
    ]
else:
    chunks = re.split(r"^#{1,6}\s+", raw, flags=re.MULTILINE)
    sections = [
        {"heading": "document" if index == 0 else f"section-{index}", "text": text.strip()}
        for index, text in enumerate(chunks)
        if text.strip()
    ]

sys.stdout.write(json.dumps({
    "source_name": file_path.name,
    "sections": sections,
}, ensure_ascii=False, separators=(",", ":")))
